openedx/features/philu_courseware/tests/test_competency_assessment_record_manager.py

```diff
@@ -0,0 +1,46 @@
+from django.test import TestCase
+from opaque_keys.edx.keys import UsageKey
+
+from openedx.features.philu_courseware.models import CompetencyAssessmentRecord
+from student.tests.factories import UserFactory
+
+from .factories import CompetencyAssessmentRecordFactory
+
+
+class CompetencyAssessmentRecordManagerTestCase(TestCase):
```

Comment:
Missing using tests for views and helpers. Are we going to write their tests separately?

Reply:
Actually we have to write the unit tests for custom functionality under the scope of this ticket. I have added unit tests for reverting from `CompetencyAssessmentRecord` table and `validate_problem_id` as well.

Reply:
Ok, but still the tests for `revert_user_post_assessment_attempts` API are missing, which is created in this PR

Reply:
Actually we have to write the unit tests only for custom functionality under the scope of this ticket. I have added unit tests for reverting from `CompetencyAssessmentRecord` table and `validate_problem_id` as well.